src/main/java/bio/overture/songsearch/service/AnalysisService.java

```diff
@@ -82,4 +85,58 @@ public Analysis getAnalysisById(String analysisId) {
                             .map(Optional::get)
                             .collect(toUnmodifiableList());
  }
+
+ public List<SampleMatchedAnalysisPair> getSampleMatchedAnalysisPairs(String analysisId) {
```

Comment:
I moved this code out of data fetcher and into the service layer. I think it makes more sense here since there is business logic tied to this functionality.

Reply:
:+1: